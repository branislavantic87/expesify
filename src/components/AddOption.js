import React from 'react';

export default class AddOptions extends React.Component {
    constructor(props) {
        super(props);
        this.handleAddOption = this.handleAddOption.bind(this);
        this.state = {
            error: undefined
        };
    }
    handleAddOption(e) {
        e.preventDefault();
        const option = e.target.elements.option.value.trim();
        const err = this.props.handleAddOption(option)

        this.setState(() => ({ error: err }))

        if (!err) {
            e.target.elements.option.value = '';
        }
    }
    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.handleAddOption}>
                    <input type="text" name="option" />
                    <button>Add option!</button>
                </form>
            </div>
        );
    }
}

