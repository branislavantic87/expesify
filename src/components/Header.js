import React from 'react';


const Header = () => (
    <div>
        <h1>Portfolio!</h1>
    </div>
);