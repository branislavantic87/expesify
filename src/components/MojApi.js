import React from 'react';

class MojApi extends React.Component {

    state = { dzejson: '' }

    componentWillMount() {
        fetch('http://localhost:3000/', { method: 'GET', mode: 'no-cors' })
            .then(res => res.json())
            .then(res => this.setState({ dzejson: JSON.stringify(res.statusCode) }))

    }
    render() {
        return (
            <div>
                <h1>Zdravo</h1>
                {this.state.dzejson && <p>{this.state.dzejson}</p>}
            </div>
        )
    }
}

export default MojApi;