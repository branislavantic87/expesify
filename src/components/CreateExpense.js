import React from 'react';
import { connect } from 'react-redux';
import Header from './Header';
import ExpenseForm from './ExpenseForm';
import { addExpense } from '../actions/expenses';

export class CreateExpense extends React.Component {
    onSubmit = (expense) => {
        // props.dispatch(addExpense(expense))
        this.props.onSubmit(expense)
        this.props.history.push('/')
    }
    render() {
        return (
            <div>
                <Header />
                <h1>Add Expense</h1>
                <ExpenseForm
                    onSubmit={this.onSubmit} />
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    onSubmit: (expense) => dispatch(addExpense(expense))
});

export default connect(undefined, mapDispatchToProps)(CreateExpense)