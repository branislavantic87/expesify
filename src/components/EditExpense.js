import React from 'react';
import Header from './Header';
const EditExpenses = (props) => {
    console.log(props)
    return (
        <div>
            <Header />
            Editing the expense with id of {props.match.params.id}
        </div>
    )
}

export default EditExpenses;