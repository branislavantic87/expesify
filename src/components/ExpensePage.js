import React from 'react';
import Header from './Header';

const ExpensePage = () => (
    <div>
        <Header />
        Ovo home page!
    </div>
)

export default ExpensePage;