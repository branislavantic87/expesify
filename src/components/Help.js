import React from 'react';
import Header from './Header';

const Help = () => (
    <div>
        <Header />
        Ovo je help page!
    </div>
)

export default Help;