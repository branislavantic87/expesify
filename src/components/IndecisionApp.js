import React from 'react';
import Header from './Header';
import Options from './Options';
import AddOptions from './AddOption';
import Action from './Action';
import OptionModal from './OptionModal';


export default class IndecisionApp extends React.Component {
    constructor(props) {
        super(props);
        this.handleDeleteOptions = this.handleDeleteOptions.bind(this);
        this.handlePick = this.handlePick.bind(this);
        this.handleAddOption = this.handleAddOption.bind(this);
        this.handleDeleteOne = this.handleDeleteOne.bind(this);
        this.handleModal = this.handleModal.bind(this);
        this.state = {
            options: [],
            selectedOption: undefined
        }
    }
    componentDidMount() {
        try {
            const json = localStorage.getItem('options');
            const options = JSON.parse(json);

            if (options) {
                this.setState(() => ({ options }))
            }
        } catch (e) {

        }
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevState.options.length !== this.state.options.length) {
            const json = JSON.stringify(this.state.options)
            localStorage.setItem('options', json)
            console.log('fetching data')
        }
    }

    componentWillUnmount() {
        console.log('saving data')
    }

    handleDeleteOptions() {
        this.setState(() => ({ options: [] }))
    }
    handleDeleteOne(optionToRemove) {
        this.setState((prevState) => ({
            options: prevState.options.filter((option) => optionToRemove !== option)
        }))
    }
    handlePick() {
        const randomNum = Math.floor(Math.random() * this.state.options.length)
        const option = this.state.options[randomNum]
        this.setState(() => ({selectedOption: true}))
    }
    handleModal(){
        this.setState(() => ({selectedOption: false}))
    }
    handleAddOption(option) {
        if (!option) {
            return 'Enter valid value to add Item';
        } else if (this.state.options.indexOf(option) > -1) {
            return 'Allready exist';
        }
        this.setState((prevState) => ({ options: prevState.options.concat(option) }))
    }
    render() {
        const subtitle = 'Put your life in hands of computer!';
        { this.state.options }
        return (
            <div>
                <Header subtitle={subtitle} />
                <div className= "container">
                <Action hasOptions={this.state.options.length > 0}
                    handlePick={this.handlePick}
                />
                <div className="widget">
                <Options
                    options={this.state.options}
                    handleDeleteOptions={this.handleDeleteOptions}
                    handleDeleteOne={this.handleDeleteOne} />
                <AddOptions handleAddOption={this.handleAddOption} />
                </div>
                <OptionModal 
                selectedOption={this.state.selectedOption}
                handleModal={this.handleModal}
                />
                </div>
            </div>
        );
    }
}

