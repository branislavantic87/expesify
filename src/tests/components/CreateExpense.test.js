import React from 'react';
import { shallow } from 'enzyme';
import { CreateExpense } from '../../components/CreateExpense';
import expenses from '../fixtures/expenses';

let onSubmit, history, wrapper;

beforeEach (() => {
    onSubmit = jest.fn();
    history = { push: jest.fn() }
    wrapper = shallow(<CreateExpense onSubmit={onSubmit} history={history} />)
});

test('should render CreateExpense correctly', () => {
    expect(wrapper).toMatchSnapshot()
});

test('should handle onSubmit', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[1])
    expect(history.push).toHaveBeenLastCalledWith('/')
    expect(onSubmit).toHaveBeenLastCalledWith(expenses[1])
});