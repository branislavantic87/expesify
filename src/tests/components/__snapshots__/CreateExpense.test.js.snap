// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CreateExpense correctly 1`] = `
<div>
  <Header />
  <h1>
    Add Expense
  </h1>
  <ExpenseForm
    onSubmit={[Function]}
  />
</div>
`;
